lib/dbservice_web/controllers/session_controller.ex

```diff
@@ -42,25 +42,54 @@ defmodule DbserviceWeb.SessionController do
   end
 
   def index(conn, params) do
+    sort_order = extract_sort_order(params)
+
     query =
       from m in Session,
-        order_by: [asc: m.id],
+        order_by: [{^sort_order, m.id}],
         offset: ^params["offset"],
         limit: ^params["limit"]
 
     query =
       Enum.reduce(params, query, fn {key, value}, acc ->
-        case String.to_existing_atom(key) do
-          :offset -> acc
-          :limit -> acc
-          atom -> from u in acc, where: field(u, ^atom) == ^value
+        case key do
+          "offset" ->
+            acc
+
+          "limit" ->
+            acc
+
+          "sort_order" ->
+            acc
+
+          "session_id_is_null" ->
+            apply_session_id_null_filter(value, acc)
+
+          _ ->
+            field_name = String.to_existing_atom(key)
+            from u in acc, where: field(u, ^field_name) == ^value
```

Comment:
```suggestion
                case String.to_existing_atom(key) do
          :offset ->
            acc

          :limit ->
            acc

          :sort_order ->
            acc

          :session_id_is_null ->
            apply_session_id_null_filter(value, acc)

          atom ->
            from(u in acc, where: field(u, ^atom) == ^value)
        end
      end)
```

Can you try this and check if it is working for all cases?